build.gradle.kts

```diff
@@ -64,8 +65,22 @@ subprojects {
     configurations.all {
         resolutionStrategy {
             dependencySubstitution {
-                substitute(module("com.facebook.react:react-native"))
-                    .with(module("com.facebook.react:react-native:$reactNativeVersion"))
+                if ( reactNativeMinorVersion >= 71 ) {
+                    // This substitution is based on React Native Gradle plugin.
+                    // Reference: https://t.ly/38jk
+                    substitute(module("com.facebook.react:react-android"))
+                        .with(module("com.facebook.react:react-android:$reactNativeVersion"))
+                    substitute(module("com.facebook.react:hermes-android"))
+                        .with(module("com.facebook.react:hermes-android:$reactNativeVersion"))
+                    // For backward-compatibility, we also substitute `react-native` module
+                    // with the new module `react-android`.
+                    substitute(module("com.facebook.react:react-native"))
+                        .with(module("com.facebook.react:react-android:$reactNativeVersion"))
```

Comment:
This follows the process performed by the React Native Gradle plugin ([reference](https://github.com/facebook/react-native/blob/ea4724b37c9f78bd33daab547d6cc4f8b7f7dd81/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt#L45-L74), [docs](https://reactnative.dev/docs/react-native-gradle-plugin#what-is-the-plugin-doing-under-the-hood)). Additionally, as shared in [this proposal](https://github.com/react-native-community/discussions-and-proposals/blob/main/proposals/0508-out-of-npm-artifacts.md#implemented-solution), now the React Native and Hermes modules are published to Maven, similar to what we do in [“react-native-mirror-publisher” repository](https://github.com/wordpress-mobile/react-native-mirror-publisher).